app/resources/linked_profile_resource.rb

```diff
@@ -0,0 +1,20 @@
+class LinkedProfileResource < BaseResource
+  attribute :token
+  attributes :external_user_id, :url, :share_to, :share_from
+
+  has_one :user
+  has_one :linked_site
+
+  filter :self, apply: -> (records, _v, options) {
+    current_user = options[:context][:user]
+    records.where(id: current_user&.id) || User.none
+  }
```

Comment:
You don't need a `self` filter for this, the client has the user_id and can use it for querying (or can just request `include=linkedProfile` on a user)